src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModel.java

```diff
@@ -525,109 +489,146 @@ int sumMismatchingQualities(final byte[] readBases,
      * @param useCachedResults if false, ignore cached results for informative indel sizes (useful for debugging)
      * @return true if read can eliminate the possibility that there's an indel of size <= maxIndelSize segregating at refStart
      */
-    @VisibleForTesting
-    boolean isReadInformativeAboutIndelsOfSize(final GATKRead read,
+    private static boolean isReadInformativeAboutIndelsOfSize(final GATKRead read,
                                                final int readStart,
                                                final byte[] refBases,
                                                final int refStart,
                                                final int maxIndelSize,
                                                final boolean useCachedResults) {
-        BitSet cachedResult = (BitSet) ((SAMRecordToGATKReadAdapter)read).getTransientAttribute("IDL");
+        BitSet cachedResult = (BitSet) ((SAMRecordToGATKReadAdapter)read).getTransientAttribute(INDEL_INFORMATIVE_BASES_CACHE_ATTRIBUTE_NAME);
         if (cachedResult == null || !useCachedResults) {
             BitSet informativeBases = new BitSet(read.getLength());
 
             // Check that we aren't so close to the end of the end of the read that we don't have to compute anything more
             if ( !(read.getLength() - readStart < maxIndelSize) && !(refBases.length - refStart < maxIndelSize) ) {
-                //TODO this value is stored for the purpose of replicating the potentially incorrect behavior of the old codepath...
+                //TODO this should be removed, see https://github.com/broadinstitute/gatk/issues/5646 to track its progress
                 final int secondaryReadBreakPosition = read.getLength() - maxIndelSize;
 
                 // We are safe to use the faster no-copy versions of getBases and getBaseQualities here,
                 // since we're not modifying the returned arrays in any way. This makes a small difference
                 // in the HaplotypeCaller profile, since this method is a major hotspot.
                 final Pair<byte[], byte[]> readBasesAndBaseQualities = AlignmentUtils.getBasesAndBaseQualitiesAlignedOneToOne(read);  //calls getBasesNoCopy if CIGAR is all match
```

Comment:
lets deconstruct this pair into readBases and readQualities upfront so there's no remember what left and right mean downstream